rotkehlchen/globaldb/handler.py

```diff
@@ -1011,6 +1011,108 @@ def add_historical_prices(entries: List['HistoricalPrice']) -> None:
 
         connection.commit()
 
+    @staticmethod
+    def add_single_historical_price(entry: HistoricalPrice) -> None:
+        """Adds the given historical price entries in the DB"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        try:
+            serialized = entry.serialize_for_db()
+            cursor.execute(
+                """INSERT OR IGNORE INTO price_history(
+                from_asset, to_asset, source_type, timestamp, price
+                ) VALUES (?, ?, ?, ?, ?)
+                """,
+                serialized,
+            )
+        except sqlite3.IntegrityError as e:
+            connection.rollback()
+            log.error(
+                f'Failed to add single historical price. {str(e)}. ',
+            )
+        # success
+        connection.commit()
+
+    @staticmethod
+    def get_manual_prices(
+        from_asset: Optional[Asset],
+        to_asset: Optional[Asset],
+    ) -> List[Dict[str, str]]:
+        """Returns prices added to the database by the user for an asset"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        querystr = (
+            'SELECT from_asset, to_asset, source_type, timestamp, price FROM price_history '
+            'WHERE source_type="A" '
+        )
+        params = []
+        if from_asset is not None:
+            querystr += 'AND from_asset=? '
+            params.append(from_asset.identifier)
+        if to_asset is not None:
+            querystr += 'AND to_asset=? '
+            params.append(to_asset.identifier)
+        querystr += 'ORDER BY timestamp'
+        query = cursor.execute(querystr, tuple(params))
+        result = query.fetchall()
+        return [
+            {
+                'from_asset': entry[0],
+                'to_asset': entry[1],
+                'timestamp': entry[3],
+                'price': entry[4],
+            }
+            for entry in result
+        ]
```

Comment:
Okay so here there is a few problems.

1. It's an easy mistake to make and I also sometimes forget about it. But you are iterating the results twice! `query.fetchall()`, iterates through all the results of the cursor and returns them in `result`. And then you iterate result again.
2. You could use the `deserialize_from_db()` method of `HistoricalPrice` and then serialize again in json without the source oracle. But ... perhaps for now doing what you do is fine, since this function is used nowhere else. But if it ends up getting used elsewhere we would probably need to change this.

So probably ignore (2)